javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -563,7 +563,7 @@ private Vector(BitMappedTrie<T> trie) {
     }
 
     @Override
-    public Vector<T> append(T element) { return appendAll(Iterator.of(element)); }
+    public Vector<T> append(T element) { return appendAll(List.of(element)); }
```

Comment:
Previously the `Iterator` was changed to a `List` anyway, at least now we're avoiding an instance creation. Also, it's quite possible that this is simply allocated on the stack, i.e. escape analysis.
If not, people can still use the bulk operation anyway, that's what it's for :).